### PR TITLE
Port of the existing module stock_orderpoint_creator for 7.0.

### PR DESCRIPTION
**Pull request by guewen**
Modified:
- view to be valid in 7.0,
- removed the redefinition of the field product_id, since the product's UOM is verified by a constraint now in procurement; it's still there but mandatory.

PR moved from Launchpad: https://code.launchpad.net/~camptocamp/stock-logistic-warehouse/7.0-port-stock-orderpoint-creator-mdh/+merge/220466 on behalf of @mdietrichc2c

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I prefer _stock_orderpoint_generator_ as name, because creator can be also the user that creates the orderpoint.

**Comment by guewen** [comment]
I only moved the PR, so @mdietrichc2c is welcome. But AFAIK, this is a port of an existing module that was in the 6.1 branch, so I don't know if it is so good to change its name (but in a sense, if it has not been ported before, the renaming should not hurt so much).

**Comment by gurneyalex** [comment]
I'm in favor of the renaming

**Comment by guewen** [comment]
I renamed the module

**Comment by pedrobaeza** [comment]
:+1: Thanks

File: stock_orderpoint_generator/__openerp__.py [hunk 1]
```diff
@@ -19,25 +19,34 @@
 #
 ##############################################################################
 
-{'name': 'Configuration of order point in mass',
+{'name': 'Order point generator',
+ 'summary': 'Configuration of order point in mass',
```
Comment (gurneyalex) [inline stock_orderpoint_generator/__openerp__.py]: "Mass configuration of stock order points"

File: stock_orderpoint_generator/__openerp__.py [hunk 2]
```diff
@@ -19,25 +19,34 @@
...
+ 'license': 'AGPL-3',
+ 'complexity': 'easy',
+ 'website': "http://www.camptocamp.com",
  'depends': ['procurement'],
  'description': """
-Add a wizard to configure massively order points for multiple product""",
+Order point generator
+=====================
+
+Add a wizard to configure massively order points for multiple products.
```
Comment (gurneyalex) [inline stock_orderpoint_generator/__openerp__.py]: "...to configure order points for multiple products in one go".

File: stock_orderpoint_generator/__openerp__.py [hunk 3]
```diff
@@ -19,25 +19,34 @@
...
+
+Contributors
+------------
+
+ * Yannick Vaucher <yannick.vaucher@camptocamp.com>
+ * Matthieu Dietrich <matthieu.dietrich@camptocamp.com>
+
+""",
  'website': 'http://www.openerp.com',
```
Comment (gurneyalex) [inline stock_orderpoint_generator/__openerp__.py]: to remove, or use OCA site.

File: stock_orderpoint_generator/__openerp__.py [hunk 4]
```diff
@@ -19,25 +19,34 @@
...
- 'init_xml': [],
- 'update_xml': ["wizard/orderpoint_creator_view.xml", "security/ir.model.access.csv"],
- 'demo_xml': [],
+ 'demo': [],
+ 'data': ["wizard/orderpoint_generator_view.xml",
+          "security/ir.model.access.csv",
+          ],
  'test': [],
```
Comment (gurneyalex) [inline stock_orderpoint_generator/__openerp__.py]: no test at all?

File: stock_orderpoint_generator/base_product_config_template.py [hunk 5]
```diff
@@ -33,42 +33,41 @@ def _get_model(self):
...
+    def create_instances(self, cr, uid, template_br,
                          product_ids, context=None):
         """ Create instances of model using template inherited model """
         if not isinstance(product_ids, list):
             product_ids = [product_ids]
 
-        # data = self.copy_data(cursor, uid, template_br.id, context=context)
-        # copy data will not work in all case and may retrieve erronus value
+        # not using self.copy_data(cr, uid, template_br.id, context=context)
+        # as copy data will not work in all case and may retrieve erronus value
```
Comment (gurneyalex) [inline stock_orderpoint_generator/base_product_config_template.py]: "return erroneous"

(I'm curious about this, though...)

File: stock_orderpoint_generator/orderpoint_template.py [hunk 6]
```diff
@@ -24,29 +24,51 @@
 from openerp.osv.orm import Model, fields
 from base_product_config_template import BaseProductConfigTemplate
 
+
 class OrderpointTemplate(BaseProductConfigTemplate, Model):
-    """ Template for orderpoints """
+    """ Template for orderpoints
+
+    Here we use same model as stock.warhouse.orderpoint but set product_id
```
Comment (gurneyalex) [inline stock_orderpoint_generator/orderpoint_template.py]: s/warhouse/warehouse/

File: stock_orderpoint_generator/orderpoint_template.py [hunk 7]
```diff
@@ -24,29 +24,51 @@
...
+    as non mandatory as we cannot remove it. This field will be ignored.
+
+    This has the advantage to ensure orderpoint and orderpoint template have
```
Comment (gurneyalex) [inline stock_orderpoint_generator/orderpoint_template.py]: "...the advantage of ensuring that the order point and the order point template have the..."

File: stock_orderpoint_generator/orderpoint_template.py [hunk 8]
```diff
@@ -24,29 +24,51 @@
...
+    same fields.
+
+    _table is redifined to separate templates from orderpoints
```
Comment (gurneyalex) [inline stock_orderpoint_generator/orderpoint_template.py]: redefined

**Comment by gurneyalex** [comment]
@mdietrichc2c some changes are needed. Most are trivial. A small test would be very welcome.

**Comment by mdietrichc2c** [comment]
@gurneyalex okay, I corrected the comments and added a basic unit test.

**Comment by gurneyalex** [comment]
:+1:

**Pull request merged**